src/build/env.py

```diff
@@ -89,5 +89,12 @@ def install(self, requirements):  # type: (Iterable[str]) -> None
 
         subprocess.check_call([sys.executable, '-m', 'ensurepip'], cwd=self._path)
 
-        cmd = [sys.executable, '-m', 'pip', 'install', '--ignore-installed', '--prefix', self._path] + list(requirements)
-        subprocess.check_call(cmd)
+        with tempfile.NamedTemporaryFile('w+', prefix='build-reqs-', suffix='.txt') as req_file:
+            req_file.write(os.linesep.join(requirements))
+            req_file.flush()
+            cmd = [
+                sys.executable, '-m', 'pip',
+                'install', '--ignore-installed', '--prefix',
+                self._path, '-r', os.path.abspath(req_file.name)
+            ]
+            subprocess.check_call(cmd)
```

Comment:
Why do we need here a requirements.txt file and not just pass the dependencies as arguments?

Reply:
I guess I should have this in the comment message. Because pip install something does not support environment markers, such as `python_version`, which is exactly what I am fixing here.

Reply:
> `pip install something` does not support environment markers

I never experienced this, @pf_moore do we have a pip issue for this?

Reply:
I'm not sure why we should. I wouldn't expect to support environment markers on the command line, it seems like unnecessary generalisation to me. But I guess someone could raise a pip feature request if they felt like it (and had a compelling enough use case - I don't think this usage would count, using a requirements file seems fine if you want to support a wider range of command line forms than pip does).

Reply:
How does pip handle environment markers in PEP-518? Does it passes it in a requirement file?

Reply:
Honestly, I don't know for sure, I'd have to look at the code.

Reply:
The code in `build_env.py` doesn't do anything special. If you can produce a test case that demonstrated the issue, maybe it should be raised as a bug... (I'd be inclined to fix it by using a requirements file, though 😉)

Reply:
Guess we can proceed with this as it is then @FFY00